doc/design.txt

```diff
@@ -210,15 +210,15 @@ An image trailer has the following structure:
      0                   1                   2                   3
```

Comment:
You're doing permutation, while it would be more correct to consider it like a tree of states, `if magic is unset do nothing`, `if magic is set check the other flag` and so on and so forth.